aquadoggo/src/network/config.rs

```diff
@@ -74,6 +74,19 @@ pub struct NetworkConfiguration {
     /// QUIC transport port.
     pub quic_port: u16,
 
+    /// Relay client behaviour enabled.
+    ///
+    /// Engage in relayed connections to remote peers.
```

Comment:
I'm not sure when a peer wouldn't want to have their messages relayed if direct connection is impossible. Is there a use case in mind for this being optional?

Reply:
If the `autonat` protocol gave us an accurate and precise indication of reachability then I'd say that `relay_client` would not need to be optional. However, it is currently quite a useful "switch" to ensure that we only listen on a relayed address and register that address with a rendezvous server if we're in relay client mode. Does that make sense? Could definitely change in the future as `autonat` improves.